src/pudl/transform/ferc1.py

```diff
@@ -853,6 +853,533 @@ def strip_col_of_non_ints(self, col: pd.Series) -> pd.DataFrame:
+        coal_plant1, coal_plant2, wind_turbine1], this algorythem will think that wind
+        turbine is steam. Ideally (also usually) if they label one, they will label all.
+        The ideal version is:
+
+        [STEAM, coal_plant1, coal_plant2, WIND, wind_turbine]. Right now, this function
+        puts the new header plant type into a column called plant_type_2. This is so we
+        can compare against the current plant_type column for accuracy and validation
+        purposes.
+        """
+        logger.info(f"{self.table_id.value}: Mapping header fuels to relevant rows")
+
+        def expand_dict(dic):
+            """Change format of header_labels.
+
+            Right now, the header_labels dictionaries defined above follow this format:
+
+            {clean_name_A: [ bad_name_A1, bad_name_A2, ...], clean_name_B: [bad_name_B1,
+            bad_nameB2]}. This is convenient visually, but it makes more sense to format
+            it like this: {bad_name_A1: clean_name_A, bad_name_A2: clean_name_A,
+            bad_name_B1: clean_name_B, bad_name_B2: clean_name_B}. We could reformat the
+            dictionaries, but for now I created this function to do it where necessariy
+            in certain functions.
+            """
+            d = {}
+            for k, lst in dic.items():
+                for i in range(len(lst)):
+                    d[lst[i]] = k
+            return d
+
+        header_labels = {  # Add these to the PLANT_CATEGORIES param ideally
+            "hydroelectric": ["hydro", "hyrdo"],
+            "internal combustion": ["internal", "interal", "international combustion"],
+            "combustion turbine": ["combustion turbine"],
+            "combined cycle": ["combined cycle"],
+            "gas turbine": ["gas"],
+            "petroleum liquids": ["oil", "diesel", "diesal"],
+            "solar": ["solar", "photovoltaic"],
+            "wind": ["wind"],
+            "geothermal": ["geothermal"],
+            "waste": ["waste", "landfill"],
+            "steam": ["steam"],
+            "nuclear": ["nuclear"],
+            "fuel_cell": ["fuel cell"],
+            "other": ["other"],
+            "renewables": ["renewables"],
+        }
+        # Clean header names
+        df["header_clean"] = np.nan
+        d = expand_dict(header_labels)
+        # Map cleaned header names onto df in a new column
+        df.loc[df["row_type"] == "header", "header_clean"] = (
+            df["plant_name_ferc1"]
+            .str.extract(rf"({'|'.join(d.keys())})", expand=False)
+            .map(d)
+        )
+        # Make groups based on utility, year, and header
+        header_groups = df.groupby(
+            [
+                "utility_id_ferc1",
+                "report_year",
+                (df["row_type"] == "header").cumsum(),
+            ]
+        )
+        # Forward fill based on headers
+        df["plant_type_2"] = np.nan
```

Comment:
Once we figure out how to handle the headers, we can get rid of this column and just add the values into either `plant_type` or `fuel_type`